### DISCUSSION TITLE
상호작용 버튼을 만들고싶어요! [치자피즈]

### DISCUSSION THREAD

**Discussion by x4cf** [naverz/zepeto-studio-kor#157] [3. Creator Samples]
메타버스 콘탠츠는 역시 상호작용 버튼이죠! 

1. 비콘에 충돌하면 오픈하고 끌 스위치 UI
2. UI Button 을 클릭하면 실행할 애니메이션 클립
3. 스타트 부분에서 버튼에 기능을 추가합니다. 자세한 내용은 아래의 링크를 참고해주세요 :)
https://studio.zepeto.me/guides/ui-events
4. 트리거 엔터와 익스트로 UI의 스위치를 제어합니다. 

<img width="856" alt="스크린샷 2022-01-02 오전 11 49 02" src="https://user-images.githubusercontent.com/90679926/147864816-346287aa-33e6-46bb-a192-c1dc3aa0f1cd.png">

아래는 완성된 스크립트의 시연 장면입니다.

1. 비콘 밖에서 비콘을 바라보는 장면
2. 비콘에 충돌해서 버튼이 활성화된 장면
3. 버튼을 누르고 제스쳐가 실행되는 장면

![Untitled](https://user-images.githubusercontent.com/90679926/147864820-da1a4d59-b332-46f0-b534-778deffd3a52.png)

여러분들은 어떻게 개발하고계신가요? 위에 제시된 방법을 제외하고도 좋은 정보가있다면 알려주세요 XD

틀린 정보가 있다면 댓글달아주세요 :D

**Comment by rjc90** [comment]
궁금한게 있습니다!
해당 기능 중, 아바타 가 모션을 취했을 때 멈추는 방법은 어떻게 하나요?

Button 눌렀을 때, 아바타가 애니메이션을 계속해서 출력합니다 ㅠ 

애니메이션은 animation Clip 선택하는 것 중에 기본 제공 하는 거 선택하였습니다!

**Reply by x4cf** [reply]
SetGesture 로 애니메이션을 실행했다면 CancleGesture 로 취소할 수 있습니다. 
코루틴 사용법을 모른다면 알려드릴게요 :D

**Reply by rjc90** [reply]
혹시 알려 주실수 있나요 ㅠㅠ

**Reply by rjc90** [reply]
![image](https://user-images.githubusercontent.com/29249173/154913598-321841a0-7a52-43f5-abd1-158385b9384d.png)


딱! 위에 써주신 여기까지 에서 멈춰있습니다!

**Reply by x4cf** [reply]
언제든 편하게 찾아주세요 :D

<img width="964" alt="스크린샷 2022-02-21 오후 5 24 18" src="https://user-images.githubusercontent.com/90679926/154916025-fbd970aa-98cc-4d25-8f83-05f5e6a8126a.png">

**Reply by rjc90** [reply]
감사합니다! 바로 테스트 해보겠습니당!

**Reply by rjc90** [reply]
해보니까 잘됩니당! 감사합니다! ㅎㅎ
